spec/porky_lib/unencrypted/file_service_spec.rb

```diff
@@ -60,100 +61,209 @@ def write_test_file(data)
   end
 
   describe '#write' do
-    it 'write plaintext data to S3' do
-      file_key = file_service.write(plaintext_data, bucket_name)
-      expect(file_key).not_to be_nil
+    it 'raises FileServiceError when file is nil' do
+      expect do
+        file_service.write(nil, bucket_name)
+      end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
     end
 
-    it 'write large plaintext data to S3' do
-      file_key = file_service.write(File.read("spec#{File::SEPARATOR}porky_lib#{File::SEPARATOR}data#{File::SEPARATOR}large_plaintext"),
-                                    bucket_name)
-      expect(file_key).not_to be_nil
+    it 'raises FileServiceError when bucket name is nil' do
+      expect do
+        file_service.write(plaintext_data, nil)
+      end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
     end
 
-    it 'write plaintext data to S3 with metadata' do
-      metadata = { content_type: 'test/data' }
-      file_key = file_service.write(plaintext_data, bucket_name, metadata: metadata)
+    it 'writes the right content to S3 if file object is used' do
+      allow(Aws::S3::Object).to receive(:new).and_return(aws_s3_object)
+      allow(aws_s3_object).to receive(:upload_file)
+
+      file = write_test_file(plaintext_data)
+      file_service.write(file, bucket_name)
+
+      expect(aws_s3_object).to have_received(:upload_file) do |tempfile_path|
+        expect(File.read(tempfile_path)).to eq(plaintext_data)
+      end
+    end
+
+    it 'writes the right content to S3 if path is used' do
+      allow(Aws::S3::Object).to receive(:new).and_return(aws_s3_object)
+      allow(aws_s3_object).to receive(:upload_file)
+
+      path = write_test_file(plaintext_data).path
+      file_service.write(path, bucket_name)
+
+      expect(aws_s3_object).to have_received(:upload_file) do |tempfile_path|
+        expect(File.read(tempfile_path)).to eq(plaintext_data)
+      end
+    end
+
+    it 'writes the right content to S3 if content is used' do
+      allow(Aws::S3::Object).to receive(:new).and_return(aws_s3_object)
+      allow(aws_s3_object).to receive(:upload_file)
+
+      file_service.write(plaintext_data, bucket_name)
+
+      expect(aws_s3_object).to have_received(:upload_file) do |tempfile_path|
+        expect(File.read(tempfile_path)).to eq(plaintext_data)
+      end
+    end
+
+    it 'handles contents containing a null byte when reading a file' do
+      null_byte_contents = "\xA0\0"
+      file_key = file_service.write(null_byte_contents, bucket_name)
       expect(file_key).not_to be_nil
```

Comment:
what if we test if it writes the right content (as the other tests above) instead of the content not being nil?
The same applies for the other null byte tests.

Reply:
I'll modify these tests to be like the ones I recently added